hackathon_site/dashboard/frontend/src/api/types.ts

```diff
@@ -42,3 +42,45 @@ export interface Category {
     max_per_team: number;
     unique_hardware_count: number;
 }
+
+/** Event API */
+export interface Profile {
+    id: number;
+    id_provided: boolean;
+    attended: boolean;
+    acknowledge_rules: boolean;
+    e_signature: string;
+    team: number;
+}
+
+interface UserWithoutProfile {
+    id: number;
+    first_name: string;
+    last_name: string;
+    email: string;
+}
+
+export interface User extends UserWithoutProfile {
+    profile: Profile;
+}
+
+export interface Team {
+    id: number;
+    team_code: string;
+    created_at: Date;
+    updated_at: Date;
+    profiles: Profile & { user: UserWithoutProfile }[];
+}
+
+/** Orders API **/
+export type OrderStatus = "Submitted" | "Ready for Pickup" | "Picked Up" | "Cancelled";
+
+export interface Order {
```

Comment:
just asking: all interfaces here are used to tell react what type of each variable is right? Does it like a header file in C/C++ ? What is the purpose of these statements?

Reply:
`interface`s (and `type`s) are the core piece of Typescript. They are type annotiations that tell you what type a variable is (which in the case of javascript, usually means what "shape" an object is).

You would use it like

```typescript
const myOrder: Order = {
  id: 5,
  hardware: [ /* Some hardware */ ],
  team: 2,
  team_code: "ABCDE",
  status: "Submitted",
  created_at: now,
  updated_at: now 
} 
```

If you didn't build the `myOrder` object properly (say you tried to put a random string in the `OrderStatus` field), then the typescript compiler (and your IDE) would throw an error. This **only** happens at compile time - typescript code ends up just getting compiled to javascript, where all of the type checking doesn't exist.

Headers in c/c++ are a bit more complicated, and are used by the compiler to link together references to different routines in different programs you compile together. Typescript is much easier to understand!